src/main/resources/schema/provider.definition.schema.v1.json

```diff
@@ -1,6 +1,6 @@
 {
     "$schema": "https://json-schema.org/draft-07/schema#",
-    "$id": "provider.definition.schema.v1.json",
+    "$id": "https://schema.cloudformation.us-east-1.amazonaws.com/provider.definition.schema.v1.json",
```

Comment:
exact value of the $id doesn't matter here, assuming we do not plan to deploy different provider definitions in different regions.

Reply:
It's possible to have regional discrepancies in a type. e.g; rolling out a new feature would mean that there's a period of time where the type is different in different regions. Also, there are occasionally longer lags on feature support into new regions that are expressed as properties on a resource. In that case, we'd see different definitions for a type in each region.

Reply:
rjlohan@ -  I think you are asking about type definitions. If so, the answer is yes, I do expect that there will be regional discrepancies - some of the types will not be available in some regions. For type definitions we will inject different $id values in each region; this way types using relative references will be validated correctly in each region. We do not have to do this for the resource definition schema because:
- resource providers do not have to reference resource definition schema at all - all our tools will validate against this schema regardless and will ignore any $schema values supplied in the resource definition
- if we release a new version of the schema it will be hosted at a different URI i.e. `provider.definition.schema.v2.json`. I (and all our tools) assume that we will not make the new definition schema public until its fully supported in all regions - exactly because all resource definitions are validated against the same schema.

Having said that, I did not have to change the $id of provider.definition.schema.v1.json at all. I think I did that because I was cross-testing schema changes using `ajv` which needs an explicit `$schema` key to function. I will probably revert this in the next commit

Reply:
Ah right, misunderstood this change.